mz_os_posix.c

```diff
@@ -21,7 +21,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
-#if defined(__APPLE__) || defined(__unix__) || defined(__riscos__)
+#if defined(__APPLE__) || defined(__unix__) || defined(__midipix__) || defined(__riscos__)
```

Comment:
This should be `#ifndef _WIN32` (or `_WINDOWS`, whatever I have used elsewhere).

Reply:
I am not sure I follow? mz_os_win32.c has system functions for win32, and mz_os_posix.c is not included when building for win32. Midipix is not a win32 target.

Reply:
max_os_posix.c is unique in that it should work on all platforms. Windows has a posix layer. By specifying `#ifndef _WIN32` it will include those files on all non-windows platforms which is essentially what is happening with `#if defined(__APPLE__) || defined(__unix__) || defined(__midipix__) || defined(__riscos__)`. So in order to prevent every non-Windows OS from having to add their own unique `defined` here we can just exclude windows.

Reply:
Alright, always nice to avoid unnecessary amounts of #ifdefs.